stdlib/filename.ml

```diff
@@ -113,8 +115,8 @@ module Unix : SYSDEPS = struct
     else
       None
 
-  let temp_dir_name =
-    try Sys.getenv "TMPDIR" with Not_found -> "/tmp"
+  let temp_dir_name = temp_dir_name ()
```

Comment:
Can you say a few words about this (seemingly unrelated) change?

Reply:
I needed to move the Windows code to C. It wasn't clear what to return from `caml_sys_temp_dir_name` for Unix, and as I needed to add an `external ...` to `filename.ml`, it seemed more appropriate to move all the code to the C side, including for Unix, rather than adding a dummy symbol for Unix.

Reply:
Are you convinced by this part of the change?

Reply:
Seems fine, yes.

Reply:
I'm slightly less convinced (just on the idea of unintended consequences) - for #12624, we decided it was OK to have "asymmetric" primitives for this kind of thing. That feels OK here, too - so just have the Unix side of the primitive return an empty string and simply not use it?

Reply:
> so just have the Unix side of the primitive return an empty string and simply not use it

Applied, thanks!